nixos/modules/profiles/hardened.nix

```diff
@@ -92,23 +92,17 @@ with lib;
   # Disable ftrace debugging
   boot.kernel.sysctl."kernel.ftrace_enabled" = mkDefault false;
 
-  # Enable reverse path filtering (that is, do not attempt to route packets
-  # that "obviously" do not belong to the iface's network; dropped packets are
-  # logged as martians).
+  # Enable strict reverse path filtering (that is, do not attempt to route
+  # packets that "obviously" do not belong to the iface's network; dropped
+  # packets are logged as martians).
   boot.kernel.sysctl."net.ipv4.conf.all.log_martians" = mkDefault true;
-  boot.kernel.sysctl."net.ipv4.conf.all.rp_filter" = mkDefault true;
+  boot.kernel.sysctl."net.ipv4.conf.all.rp_filter" = mkDefault "1";
```

Comment:
Why do we have to change this? The sysctl module maps `true` to `1` for us (since `toString true == "1"`).

Reply:
I wondered the same. I think the answer is

> In the case of rp_filter (which systemd would set to 2 (loose)), make
our overrides to "1" more explicit.

Reply:
Yes. `systemd` sets these to "2" (loose) in the snippets already, the hardenened rules set it to "1" (strict).

Having a boolean here doesn't really explain what and why we're doing.
I also updated the comment above, to add the "strict" bit.